alias to interface type

The following program `sample.go` triggers a panic:

```go
package main
  
import "fmt"

type Error interface {
    error
    Code() string
}

type MyError Error

type T struct {
    Name string
}

func (t *T) Error() string { return "err: " + t.Name }
func (t *T) Code() string  { return "code: " + t.Name }

func newT(s string) MyError { return &T{s} }

func main() {
    t := newT("foo")
    fmt.Println(t.Code())
}
```

Expected result:
```console
$ go run ./sample.go
code: foo
```

Got:
```console
$ yaegi ./sample.go
scope.go:153: nil reflect type
panic: nil reflect type

goroutine 1 [running]:
log.Panic(0xc00018d768, 0x1, 0x1)
        /usr/lib/go/src/log/log.go:338 +0xac
github.com/containous/yaegi/interp.(*scope).add(0xc0002501c0, 0xc00023f790, 0xc00024b200)
        /home/marc/go/src/github.com/containous/yaegi/interp/scope.go:153 +0x173
github.com/containous/yaegi/interp.(*Interpreter).cfg.func1(0xc00024a800, 0xc00018dd18)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:250 +0x2438
github.com/containous/yaegi/interp.(*node).Walk(0xc00024a800, 0xc00018dd18, 0xc00018dd50)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:112 +0xc9
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ec300, 0xc00018dd18, 0xc00018dd50)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:116 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc0000eea00, 0xc0000ec300, 0xc0000c49d8, 0x4, 0x0, 0x0, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:39 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000eea00, 0xc0000e46e0, 0x157, 0x357, 0xc0000e46e0, 0x157, 0x0, 0xc0002241e0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:266 +0x1ed
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:77 +0x2f9
```

--- a/_test/interface11.go
+++ b/_test/interface11.go
@@ -1,0 +1,27 @@
+package main
+
+import "fmt"
+
+type Error interface {
+	error
+	Code() string
+}
+
+type MyError Error
+
+type T struct {
+	Name string
+}
+
+func (t *T) Error() string { return "err: " + t.Name }
+func (t *T) Code() string  { return "code: " + t.Name }
+
+func newT(s string) MyError { return &T{s} }
+
+func main() {
+	t := newT("foo")
+	fmt.Println(t.Code())
+}
+
+// Output:
+// code: foo

--- a/example/pkg/pkg_test.go
+++ b/example/pkg/pkg_test.go
@@ -109,7 +109,7 @@ func TestPackagesError(t *testing.T) {
 		{
 			desc:     "different packages in the same directory",
 			goPath:   "./_pkg9/",
-			expected: "found packages pkg and pkgfalse in _pkg9/src/github.com/foo/pkg",
+			expected: "1:21: import \"github.com/foo/pkg\" error: found packages pkg and pkgfalse in _pkg9/src/github.com/foo/pkg",
 		},
 	}
 

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1115,12 +1115,12 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 			} else if ti := n.typ.lookupField(n.child[1].ident); len(ti) > 0 {
 				// Handle struct field
 				n.val = ti
-				switch n.typ.cat {
-				case interfaceT:
+				switch {
+				case isInterfaceSrc(n.typ):
 					n.typ = n.typ.fieldSeq(ti)
 					n.gen = getMethodByName
 					n.action = aMethod
-				case ptrT:
+				case n.typ.cat == ptrT:
 					n.typ = n.typ.fieldSeq(ti)
 					n.gen = getPtrIndexSeq
 					if n.typ.cat == funcT {

--- a/interp/gta.go
+++ b/interp/gta.go
@@ -156,6 +156,8 @@ func (interp *Interpreter) gta(root *node, rpath string) ([]*node, error) {
 				default: // import symbols in package namespace
 					sc.sym[name] = &symbol{kind: pkgSym, typ: &itype{cat: srcPkgT, path: ipath}}
 				}
+			} else {
+				err = n.cfgErrorf("import %q error: %s", ipath, err.Error())
 			}
 
 		case typeSpec:
@@ -165,7 +167,8 @@ func (interp *Interpreter) gta(root *node, rpath string) ([]*node, error) {
 				return false
 			}
 			if n.child[1].kind == identExpr {
-				n.typ = &itype{cat: aliasT, val: typ, name: typeName, path: rpath}
+				n.typ = &itype{cat: aliasT, val: typ, name: typeName, path: rpath, field: typ.field}
+				copy(n.typ.method, typ.method)
 			} else {
 				n.typ = typ
 				n.typ.name = typeName

--- a/interp/gta.go
+++ b/interp/gta.go
@@ -157,7 +157,7 @@ func (interp *Interpreter) gta(root *node, rpath string) ([]*node, error) {
 					sc.sym[name] = &symbol{kind: pkgSym, typ: &itype{cat: srcPkgT, path: ipath}}
 				}
 			} else {
-				err = n.cfgErrorf("import %q error: %s", ipath, err.Error())
+				err = n.cfgErrorf("import %q error: %v", ipath, err)
 			}
 
 		case typeSpec:

--- a/interp/gta.go
+++ b/interp/gta.go
@@ -167,7 +167,7 @@ func (interp *Interpreter) gta(root *node, rpath string) ([]*node, error) {
 				return false
 			}
 			if n.child[1].kind == identExpr {
-        n.typ = &itype{cat: aliasT, val: typ, name: typeName, path: rpath, field: typ.field, incomplete: typ.incomplete}
+				n.typ = &itype{cat: aliasT, val: typ, name: typeName, path: rpath, field: typ.field, incomplete: typ.incomplete}
 				copy(n.typ.method, typ.method)
 			} else {
 				n.typ = typ

--- a/interp/run.go
+++ b/interp/run.go
@@ -1274,6 +1274,12 @@ func _return(n *node) {
 		switch t := def.typ.ret[i]; t.cat {
 		case errorT:
 			values[i] = genInterfaceWrapper(c, t.TypeOf())
+		case aliasT:
+			if isInterfaceSrc(t) {
+				values[i] = genValueInterface(c)
+			} else {
+				values[i] = genValue(c)
+			}
 		case interfaceT:
 			values[i] = genValueInterface(c)
 		default:

--- a/interp/type.go
+++ b/interp/type.go
@@ -694,7 +694,7 @@ func (t *itype) lookupField(name string) []int {
 
 	for i, f := range t.field {
 		switch f.typ.cat {
-		case ptrT, structT:
+		case ptrT, structT, interfaceT, aliasT:
 			if index2 := f.typ.lookupField(name); len(index2) > 0 {
 				return append([]int{i}, index2...)
 			}
@@ -806,6 +806,8 @@ func (t *itype) refType(defined map[string]bool) reflect.Type {
 		t.val.rtype = interf
 	}
 	switch t.cat {
+	case aliasT:
+		t.rtype = t.val.refType(defined)
 	case arrayT, variadicT:
 		if t.size > 0 {
 			t.rtype = reflect.ArrayOf(t.size, t.val.refType(defined))
@@ -867,6 +869,8 @@ func (t *itype) frameType() (r reflect.Type) {
 		panic(err)
 	}
 	switch t.cat {
+	case aliasT:
+		r = t.val.frameType()
 	case arrayT, variadicT:
 		if t.size > 0 {
 			r = reflect.ArrayOf(t.size, t.val.frameType())
@@ -916,7 +920,13 @@ func isShiftOperand(n *node) bool {
 	return false
 }
 
-func isInterface(t *itype) bool { return t.cat == interfaceT || t.TypeOf().Kind() == reflect.Interface }
+func isInterfaceSrc(t *itype) bool {
+	return t.cat == interfaceT || (t.cat == aliasT && isInterfaceSrc(t.val))
+}
+
+func isInterface(t *itype) bool {
+	return isInterfaceSrc(t) || t.TypeOf().Kind() == reflect.Interface
+}
 
 func isStruct(t *itype) bool { return t.TypeOf().Kind() == reflect.Struct }
 

